integration/tests/epochs/epoch_test.go

```diff
@@ -136,31 +138,52 @@ func (s *Suite) TestEpochJoin() {
 
 	env := utils.LocalnetEnv()
 
+	role := flow.RoleAccess
 	// stake a new node
-	info := s.StakeNode(ctx, env, flow.RoleConsensus)
-
-	// get node info from staking table
-	nodeInfoCDC := s.ExecuteGetNodeInfoScript(ctx, env, info.NodeID)
-	nodeInfo, ok := nodeInfoCDC.(cadence.Struct)
-	require.True(s.T(), ok)
+	info := s.StakeNode(ctx, env, role)
+	testContainerName := fmt.Sprintf("epochs-test-join-%s-%s", info.Role, info.NodeID)
 
-	// make sure node info we generated matches what we get from the flow staking table
-	nodeIDFromState := string(nodeInfo.Fields[0].(cadence.String))
-	require.Equal(s.T(), info.NodeID.String(), nodeIDFromState, "expected node ID generated in test to equal node ID from staking table ")
+	// check if node is in proposed table
+	proposedTable := s.ExecuteGetProposedTableScript(ctx, env, info.NodeID)
+	assert.Contains(s.T(), proposedTable, info.NodeID, "expected new node to be in proposed table")
 
 	result := s.SetApprovedNodesScript(ctx, env, append(s.net.Identities().NodeIDs(), info.NodeID)...)
 	require.NoError(s.T(), result.Error)
 
-	// get new approved nodes list and make sure new node was added correctly
+	// ensure node ID in approved list
 	approvedNodes := s.ExecuteReadApprovedNodesScript(ctx, env)
+	assert.Contains(s.T(), approvedNodes, info.NodeID)
+
+	nodeConfig := testnet.NewNodeConfig(role, testnet.WithID(info.NodeID))
+	testContainerConfig := testnet.NewContainerConfig(testContainerName, nodeConfig, info.NetworkingKey, info.StakingKey)
+	err := testContainerConfig.WriteKeyFiles(s.net.BootstrapDir, flow.Localnet, info.MachineAccountAddress, encodable.MachineAccountPrivKey{PrivateKey: info.MachineAccountKey}, role)
+	require.NoError(s.T(), err)
+
+	// download root snapshot from access node, wait until we are in the epoch setup phase
+	var snapshot *inmem.Snapshot
+	for {
+		snapshot, err = s.client.GetLatestProtocolSnapshot(ctx)
+		require.NoError(s.T(), err)
 
-	found := false
-	for _, val := range approvedNodes.(cadence.Array).Values {
-		if string(val.(cadence.String)) == info.NodeID.String() {
-			found = true
+		currentPhase, err := snapshot.Phase()
+		require.NoError(s.T(), err)
+		_, err = snapshot.Identity(info.NodeID)
+
+		if currentPhase == flow.EpochPhaseSetup && err == nil {
+			break
```

Comment:
Does this reproduce the failure to see the joining node in the identity table you were mentioning?

Reply:
That currently doesn't work because node ID is never present so it loops forever. [Instead I wait for the sealed view of the first DKG phase to ensure we are in the epoch setup phase](https://github.com/onflow/flow-go/pull/1603/commits/d2ba2037f0211fc566cdbc0f9c017d877b762a48#diff-9b4499ee5310f158af66e16a74547dbdd0639c55e177ee8305040547fe0187efR174). Then when we start our node we consistently get the following err. 

> {"level":"fatal","node_role":"access","node_id":"e4607ca0daa0d9ec74f51fb0a11ce848b3ab28a6ecf1b916b89fdba09a7d6dc2","error":"identity not found (e4607ca0daa0d9ec74f51fb0a11ce848b3ab28a6ecf1b916b89fdba09a7d6dc2)","time":"2021-11-10T13:00:34Z","message":"node identity not found in the identity list of the finalized state: e4607ca0daa0d9ec74f51fb0a11ce848b3ab28a6ecf1b916b89fdba09a7d6dc2"}

[This validation in init local of the fnb is what's failing. ](https://github.com/onflow/flow-go/blob/master/cmd/scaffold.go#L709)